### DISCUSSION TITLE
Integration into Humhub tasks (sponsored)

### DISCUSSION THREAD

**Discussion by simialbi** [simialbi/yii2-kanban#5] [Ideas]
Users over at Humhub are keen for a Kanban view that can be integrated with the existing Tasks module.
humhub/humhub-modules-tasks#115

@olantrust is willing to sponsor work on it @simialbi please message ict @ olan . uk if interested.

**Comment by simialbi** [comment]
@olantrust I set up a community version of humhub. It seems that there are plenty differences between your task module and this kanban one. E.g. Task module works with lists, not boards. And there are no buckets. In addition the layout differences: Your task module has some tabbed system, while kanban has all on one modal (and so on).
Maybe I should generate a Login for you, so you can see a demo of kanban and look at kanban yourself, so you have a base to decide how'd you like to get it implemented. What do you think?

**Reply by spoorun** [reply]
@simialbi appreciate the heads-up; a login would be ideal to allow a little analysis. Thanks

**Reply by simialbi** [reply]
I sent it via mail to ict @ olan . uk

**Comment by ArchBlood** [comment]
Just seen this issue randomly, and I love the idea, although I believe it would be for the best to create a completely different module overall, similar to the Calendar and External Calendar modules, one extends the other as you'll see and enhances the other for more features, as for the kanban, even though I love the idea of using SCSS, but currently, HumHub uses both CSS and LESS methods so I'm not sure how well it would work out if one were to use SCSS in a module, something that I'd love to find out for myself 🤔

https://github.com/humhub/calendar
https://github.com/humhub-contrib/external_calendar

**Comment by simialbi** [comment]
@olantrust, @spoorun 
I started to implement a HumHub version of this repository, but noticed, it's a lot more work than expected. I have to rewrite half the module (e.g. can't use turbo and have to use humhub classes, no bs4, bs3 instead, no fa5, f4 instead and so on). That might take a lot more time than initially expected.

**Reply by ArchBlood** [reply]
If you don't mind @GreenMeteor could try our hand at doing a conversion of the yii2 package here, ideally HumHub classes and file structure aren't that different from yii2's and it would be something that we have done before in the past for select clients, the main issue that we ourselves would face would be the conversion of SCSS to full CSS & LESS structure as well as the conversion of the JavaScript to support HumHub's JavaScript API. 🤔

https://docs.humhub.org/docs/develop/javascript-index/

**Reply by simialbi** [reply]
@ArchBlood of course a little support would be appreciated. Especially because I can't spend much time for this task. I suggest that I create a new public repository with the current status of the project and invite @GreenMeteor or you (or both) to this repository?
SCSS should not be a problem, you can e.g. use https://github.com/sass/dart-sass in your IDE to compile the files on change...

**Reply by ArchBlood** [reply]
Sure thing, if you need a way of contact please don't hesitate to email me at kodo@greenmeteor.net

**Reply by simialbi** [reply]
https://github.com/simialbi/humhub-kanban

**Reply by ArchBlood** [reply]
I'll look over the current status of the module tomorrow morning and go from there then do what changes that are needed (if any are needed) as well as compile a version that is pre-built so that already used classes could still be used.
